docs/docs/adrs/adr-011-improving-test-confidence.md

```diff
@@ -0,0 +1,208 @@
+| **v3 consumer (sdk47, ibc7)**   | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ✅              |
+| **main consumer**               | ❌               | ❌           | ❌            | ❌              | ❌              | ❌            | ❌                | ❌              |
+| **neutron**                     | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ❌              |
+| **stride**                      | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ❌              |
+
+
+##### Introducing e2e CometMock tests
+
+CometMock is a simplified version of the [CometBFT](https://github.com/cometbft/cometbft) consensus engine. It supports most operations performed by CometBFT while also being lightweight and relatively easy to use.
+
+CometMock tests allow more nuanced control of test scenarios because CometMock can "fool" the blockchain app into thinking that a certain number of blocks had passed.
+**This allows us to test very nuanced scenarios, difficult edge cases and long-running operations (such as unbonding operations).**
+
+Examples of tests made easier with CometMock are listed below:
+- regression tests
+- non-determinism tests
+- upgrade tests
+- state-breaking changes
+
+With CometMock, the **matrix test** approach can also be used. The image below illustrates a CometMock setup that can be used to discover non-deterministic behavior and state-breaking changes.
+![e2e matrix tests](../../figures/cometmock_matrix_test.png)
+
+This table explains which versions are tested against each other for the same set of test traces:
+* ✅ marks a passing test
+* ❌ marks a failing test
+
+| **SCENARIO**                    | **start chain** | **add key** | **delegate** | **undelegate** | **redelegate** | **downtime** | **equivocation** | **stop chain** |
+|---------------------------------|-----------------|-------------|--------------|----------------|----------------|--------------|------------------|----------------|
+| **v3 provi + v3 consu**         | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ✅              |
+| **main provi + main consu**     | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ✅              |
+| **commit provi + commit consu** | ✅               | ❌           | ✅            | ❌              | ✅              | ✅            | ❌                | ❌              |
+
+
+
+Briefly; multiple versions of the application are run against the same CometMock instance and any deviations in app behavior would result in `app hash` errors (the apps would be in different states after performing the same set of actions).
+
+#### 3. Introduce innovative testing approaches
```

Comment:
The headline level doesn't match the one of the other numbered headlines. What's the purpose of the numbered headlines?

Reply:
Outlining how many points to address there are